src/routes/event.tsx

```diff
@@ -49,47 +58,105 @@ const noMatchesStyle = css`
   margin: -0.35rem 0 0;
 `
 
+const eventErrorStyle = css`
+  justify-content: center;
+  align-items: start;
+  padding: ${spacing};
+  margin-top: 0.5rem;
+
+  @media (min-width: 800px) {
+    grid-template-columns: minmax(20rem, 25rem) 21rem;
+  }
+`
+
+const networkErrorStyle = css`
+  grid-template-columns: minmax(21rem, 23rem);
+  justify-content: center;
+  align-items: start;
+  grid-gap: ${spacing};
+  padding: ${spacing};
+  margin-top: 0.5rem;
+
+  @media (min-width: 800px) {
+    grid-template-columns: minmax(20rem, 25rem) 21rem;
+  }
+`
+
 const Event = ({ eventKey }: Props) => {
   const matches = useEventMatches(eventKey)
   const eventInfo = useEventInfo(eventKey)
-  const newestIncompleteMatch = matches && nextIncompleteMatch(matches)
-
+  const newestIncompleteMatch = isData(matches) && nextIncompleteMatch(matches)
   return (
     <Page
-      name={eventInfo?.name || <code>{eventKey}</code>}
+      name={isData(eventInfo) ? eventInfo.name : <code>{eventKey}</code>}
       back="/"
-      class={eventStyle}
+      class={
+        eventInfo instanceof Error
+          ? eventInfo instanceof NetworkError
+            ? networkErrorStyle
+            : eventErrorStyle
+          : eventStyle
```

Comment:
We can simplify this so that the ternaries aren't nested:

```suggestion
        eventInfo instanceof NetworkError
          ? networkErrorStyle
          : eventInfo instanceof Error
          ? eventErrorStyle
          : eventStyle
```